src/app/layouts/navbar.component.ts

```diff
@@ -27,6 +32,12 @@ export class NavbarComponent implements OnInit {
         if (this.name == null || this.name.trim() === '') {
             this.name = this.email;
         }
+
+        // Ensure provides are loaded
+        if (await this.syncService.getLastSync() == null) {
+            await this.syncService.fullSync(false);
+        }
+        this.providers = await this.userService.getAllProviders();
```

Comment:
Could this happen perhaps after init, I know you mentioned a reactive model which would be great, but seems like firing this off asynchronously after paint/render and reactively showing the navigation would be a better approach than blocking the rendering of the component itself on init.

Reply:
We don't block of rendering. `OnInit` does not await the promise before rendering.